src/main/java/de/retest/web/selenium/WebElementWrapper.java

```diff
@@ -0,0 +1,131 @@
+package de.retest.web.selenium;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.Dimension;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.Point;
+import org.openqa.selenium.Rectangle;
+import org.openqa.selenium.WebDriverException;
+import org.openqa.selenium.WebElement;
+
+public class WebElementWrapper implements WebElement {
+
+	private final WebElement delegate;
+	private final AutocheckingRecheckDriver driver;
+
+	public WebElementWrapper( final WebElement delegate, final AutocheckingRecheckDriver driver ) {
+		if ( delegate instanceof WebElementWrapper ) {
+			throw new IllegalStateException( "Wrapping doubly!" );
+		}
+		this.delegate = delegate;
+		this.driver = driver;
+	}
+
+	@Override
+	public <X> X getScreenshotAs( final OutputType<X> target ) throws WebDriverException {
+		return delegate.getScreenshotAs( target );
+	}
+
+	protected static String toStep( final String action, final WebElement delegate ) {
+		String result = delegate.toString();
+		if ( result.contains( "->" ) ) {
+			// remove driver info
+			result = result.substring( result.lastIndexOf( "->" ) + 2 ).trim();
+			// remove trailing ]
+			result = result.substring( 0, result.length() - 1 );
+			// remove identification criterion (e.g. id, class, ...)
+			result = result.substring( result.lastIndexOf( ':' ) + 1 ).trim();
+			return action + "_" + result;
+		}
+		return result;
+	}
+
+	@Override
+	public void click() {
+		driver.check( toStep( "click", delegate ) );
+		delegate.click();
+	}
+
+	@Override
+	public void submit() {
+		driver.check( toStep( "submit", delegate ) );
+		delegate.submit();
+	}
+
+	@Override
+	public void sendKeys( final CharSequence... keysToSend ) {
+		driver.check( toStep( "enter", delegate ) );
```

Comment:
I think it would be helpful if the step would also include the `keysToSend`, what do you think? Potentially too much chars then?

Reply:
We can shorten to e.g. 10 chars?